File: backend/model/shipment.py
from typing import List, Optional
from enum import Enum
from pydantic import BaseModel
from datetime import datetime


class ShipmentStats(BaseModel):
    label: str
    value: str | int
    color: str


class EventCategory(str, Enum):
    PICK_UP = "pick_up"
    DEPARTURE = "departure"
    DELAY = "delay"
    MEX_INSPECCION = "mex_inspeccion"
    CLEAR_MEX = "clear_mex"
    USA_INSPECCION = "usa_inspeccion"
    CLEAR_USA = "clear_usa"
    SAFETY_YARD = "safety_yard"
    DELIVERY = "deliver"


class ShipmentEvent(BaseModel):
    category: EventCategory
    dateTime: datetime
    notes: Optional[str] = None


class ShipmentCreate(BaseModel):
    tracking_number: str
    costumer_tracking: str
    cliente: str
    truck: str
    vehicle_type: str
    trailer: str
    origen: str
    destino: str
    type_operation: str


class Shipment(ShipmentCreate):
    id: int
    status: str
    events: List[ShipmentEvent]

File: Frontend/src/types/Shipment.ts


export interface Shipment{
    id : number;
    trcking_Number : string;
    costumer_tracking : string;
    cliente:string;
    truck:string;
    vehicleType : string;
    trailer : string;
    orgien:string,
    destino:string,
    type_operation:string,
    status:string;
    events: ShipmentEvent[];


}

export interface ShipmentEvent{
    category:EventCategory;
    dateTime:Date|null;
    notes?:string;
}
export type EventCategory = 
  | "pick_up" 
  | "departure" 
  | "delay"
  | "mex_inspeccion" 
  | "clear_mex" 
  | "usa_inspeccion" 
  | "clear_usa"
  |"safety_yard" 
  | "deliver";


export interface ShipmentCreate{

    trcking_Number : string;
    costumer_tracking : string;
    cliente:string;
    truck:string;
    vehicleType : string;
    trailer : string;
    orgien:string,
    destino:string,
    type_operation:string,

}